### PR TITLE
feat: show external jobs on the resource pool page

### PR DESCRIPTION
**Pull request by jagadeesh545**
### Display external jobs on the Resource Pool page Jobs table.
Currently, DeterminedAI WebUI only shows DeterminedAI jobs on the Resource Pool page's job table. But the customers want to see even the external jobs on the jobs table. This PR implements the Web UI changes for this feature. 

1. Reused the Icon for `Group` as the icon for `External Job`. Also, the tooltip for the icon will show the text `External`.
2. When displaying the user icon on the Jobs table, there might not be a DetailedUser instance loaded in the DeterminedAI's `users` list for external jobs. Updated the code to create a new DetailedUser instance for the external user. 

## Test Plan
Tested these changes manually as part of the EE PR determined-ai/determined-ee#855.
Refer images below.
image

## Checklist
- [X] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
[FE-23](https://hpe-aiatscale.atlassian.net/browse/FE-23)

[FE-23]: https://hpe-aiatscale.atlassian.net/browse/FE-23?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/utils/job.ts [hunk 1]
```diff
@@ -14,6 +14,9 @@ export const unsupportedQPosSchedulers = new Set<Api.V1SchedulerType>([
 ]);
 
 export const jobTypeIconName = (jobType: JobType): IconName => {
+  if (jobType === JobType.EXTERNAL) {
+    return 'external';
+  }
```
Comment (hkang1) [inline webui/react/src/utils/job.ts:19]: **question:** should `external` be added to `CommandType`? Is this for the detached mode job?

**Review by hkang1 — COMMENTED** [review]
Overall looks great! Left a couple of comment & Qs.

File: webui/react/src/utils/job.ts (hunk 1 above)
Comment (jagadeesh545) [reply webui/react/src/utils/job.ts:19]: External jobs are jobs that are not part of DeterminedAI. For example, in an HPC cluster, jobs running on SLURM Workload Manager that are not launched DeterminedAI will be treated as external jobs. So, `external` will not be part of `CommandType` which includes non-experiment DeterminedAI-created job types.

File: webui/react/src/pages/JobQueue/JobQueue.tsx [hunk 2]
```diff
@@ -316,9 +325,20 @@ const JobQueue: React.FC<Props> = ({ selectedRp, jobState }) => {
...
               ...col,
-              render: createOmitableRenderer<Job, FullJob>('entityId', (_, r) =>
-                userRenderer(users.find((u) => u.id === r.userId)),
-              ),
+              render: createOmitableRenderer<Job, FullJob>('entityId', (_, r) => {
+                let user = users.find((u) => u.id === r.userId);
+                if (!user) {
+                  // This is an external user. Create a new DetailedUser instance.
+                  const externalUser: DetailedUser = {
+                    id: Date.now(),
```
[the remark this answers is not in the document]
Comment (jagadeesh545) [reply webui/react/src/pages/JobQueue/JobQueue.tsx]: Sounds good. Updated as suggested.

**Review by hkang1 — APPROVED** [review]
LGTM! Thanks for the updates

**Pull request merged**